system/Entity/Entity.php

```diff
@@ -0,0 +1,613 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Entity;
+
+use CodeIgniter\Entity\Cast\ArrayCast;
+use CodeIgniter\Entity\Cast\BooleanCast;
+use CodeIgniter\Entity\Cast\CastInterface;
+use CodeIgniter\Entity\Cast\CSVCast;
+use CodeIgniter\Entity\Cast\DatetimeCast;
+use CodeIgniter\Entity\Cast\FloatCast;
+use CodeIgniter\Entity\Cast\IntegerCast;
+use CodeIgniter\Entity\Cast\JsonCast;
+use CodeIgniter\Entity\Cast\ObjectCast;
+use CodeIgniter\Entity\Cast\StringCast;
+use CodeIgniter\Entity\Cast\TimestampCast;
+use CodeIgniter\Entity\Exceptions\CastException;
+use CodeIgniter\I18n\Time;
+use Exception;
+use JsonSerializable;
+
+/**
+ * Entity encapsulation, for use with CodeIgniter\Model
+ */
+class Entity implements JsonSerializable
```

Comment:
I would presume you didn't change anything here aside from changing the namespace. It'd be hard to check the contents of this large diff.

Reply:
Only 
		$keys = array_filter(array_keys($this->attributes), function ($key) {
			return strpos($key, '_') !== 0;
		});
instead of:
$keys = array_keys($this->attributes);
$keys = array_filter($keys, function ($key) {
	return strpos($key, '_') !== 0;
});

and this: 
$keys = array_unique(array_merge(array_diff($keys, $this->datamap), array_keys($this->datamap)));

instead of this: 
$keys = array_keys($this->datamap);
$keys = array_unique(array_merge(array_diff($keys, $this->datamap), $keys));

last thing this: 
$result = $this->castAs($result, $key);

instead of this:
$result = $this->castAs($result, $key, 'get');
where last param is default to 'get'

and replaced === null check with is_null method

that's all